android/java/org/chromium/chrome/browser/rewards/tipping/PopupWindowTippingTabletUI.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package org.chromium.chrome.browser.rewards.tipping;
+
+import android.content.res.Configuration;
+import android.content.res.Resources;
+import android.graphics.Color;
+import android.graphics.drawable.ColorDrawable;
+import android.os.Bundle;
+import android.text.SpannableString;
+import android.text.method.LinkMovementMethod;
+import android.util.DisplayMetrics;
+import android.view.Gravity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
+import android.view.WindowManager;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.fragment.app.DialogFragment;
+
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.BraveRewardsHelper;
+import org.chromium.chrome.browser.util.ConfigurationUtils;
+
+public class PopupWindowTippingTabletUI extends DialogFragment {
+    private static final String TAG = "TippingBanner";
+    private static final String ANCHOR_Y_POSITION = "y_position";
+
+    int mCurrentTabId = -1;
+
+    public static PopupWindowTippingTabletUI newInstance(View anchorView, int currentTabId) {
+        int[] location = new int[2];
+        anchorView.getLocationOnScreen(location);
+        int popup_y_position = location[1] + anchorView.getHeight() - 40;
+        PopupWindowTippingTabletUI fragment = new PopupWindowTippingTabletUI();
+        Bundle args = new Bundle();
+        args.putInt(RewardsTippingBannerActivity.TAB_ID_EXTRA, currentTabId);
+        args.putInt(ANCHOR_Y_POSITION, popup_y_position);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        setDialogParams();
+    }
+
+    private void setDialogParams() {
+        DisplayMetrics displayMetrics = new DisplayMetrics();
+        getActivity().getWindowManager().getDefaultDisplay().getMetrics(displayMetrics);
+        int mDeviceWidth = displayMetrics.widthPixels;
+        Window window = getDialog().getWindow();
+
+        WindowManager.LayoutParams params = window.getAttributes();
+        boolean isLandscape = ConfigurationUtils.isLandscape(getActivity());
+        if (isLandscape) {
```

Comment:
we can similar to below, 
float widthRatio = isLandscape ? 0.8 : 0.96;
params.width = (int) (widthRatio*mDeviceWidth);

Reply:
updated